src-electron/db/zap-schema.sql

```diff
@@ -3205,6 +3206,558 @@ UPDATE SESSION
+  INNER JOIN 
+    PACKAGE p 
+  ON 
+    c.PACKAGE_REF = p.PACKAGE_ID 
+  INNER JOIN 
+  	SESSION_PACKAGE spk
+  ON 
+   	p.PACKAGE_ID = spk.PACKAGE_REF
+  INNER JOIN
+    SESSION_PARTITION spt 
+  ON 
+    spk.SESSION_PARTITION_REF = spt.SESSION_PARTITION_ID
+  INNER JOIN
+    SESSION_PARTITION spt2
+  ON
+    spt.SESSION_REF = spt2.SESSION_REF
+  INNER JOIN 
+    SESSION_PACKAGE spk2 
+  ON 
+    spt2.SESSION_PARTITION_ID = spk2.SESSION_PARTITION_REF
+  INNER JOIN 
+    PACKAGE p2
+  ON 
+    spk2.PACKAGE_REF = p2.PACKAGE_ID
+  INNER JOIN
+    CLUSTER c2 
+  ON 
+    p2.PACKAGE_ID = c2.PACKAGE_REF
+  WHERE 
+    spk.SESSION_PARTITION_REF = NEW.SESSION_PARTITION_REF
+  AND 
+    spk.PACKAGE_REF = NEW.PACKAGE_REF
+  AND 
+    spk2.ENABLED = true
+  AND (
+    (c.CODE = c2.CODE AND c.MANUFACTURER_CODE = c2.MANUFACTURER_CODE)
+    OR 
+    (c.CODE = c2.CODE AND c.MANUFACTURER_CODE IS NULL AND c2.MANUFACTURER_CODE IS NULL)
+  )
+  AND 
+    p.PACKAGE_ID <> p2.PACKAGE_ID;
+END;
+
+
+/* Trigger that deals wuth code conflicts in attributes when a session package is inserted */
+CREATE TRIGGER ATTRIBUTE_CODE_CONFLICT_MESSAGE_ON_INSERT
+AFTER INSERT ON SESSION_PACKAGE
+WHEN
+  EXISTS(
+    SELECT 
+      1
+    FROM 
+      SESSION_PACKAGE spk
+    JOIN 
+      PACKAGE p 
+    ON 
+      spk.PACKAGE_REF = p.PACKAGE_ID
+    WHERE 
+      p.TYPE = 'zcl-xml-standalone'
+    AND 
+      spk.package_ref = NEW.package_ref
+    AND 
+      NEW.ENABLED = true
+  )
+BEGIN
+  INSERT INTO SESSION_NOTICE(
+    SESSION_REF, 
+    NOTICE_TYPE, 
+    NOTICE_MESSAGE, 
+    NOTICE_SEVERITY, 
+    DISPLAY, 
+    SEEN
+  )
+  SELECT 
+    spt.SESSION_REF, 
+    'ERROR', 
+    'Attribute code conflict in ' || p.PATH || ' and ' || p2.PATH || ' in attribute=' || a.CODE || ' cluster=' || a.CLUSTER_REF,
+    2, 
+    1, 
+    0
+  FROM 
+    ATTRIBUTE a
+  INNER JOIN
+    PACKAGE p 
+  ON 
+    a.PACKAGE_REF = p.PACKAGE_ID 
+  INNER JOIN 
+    SESSION_PACKAGE spk
+  ON 
+    p.PACKAGE_ID = spk.PACKAGE_REF
+  INNER JOIN 
+    SESSION_PARTITION spt
+  ON 
+    spk.SESSION_PARTITION_REF = spt.SESSION_PARTITION_ID   
+  INNER JOIN 
+    SESSION_PARTITION spt2
+  ON 
+    spt.SESSION_REF = spt2.SESSION_REF
+  INNER JOIN 
+    SESSION_PACKAGE spk2 
+  ON 
+    spt2.SESSION_PARTITION_ID = spk2.SESSION_PARTITION_REF
+  INNER JOIN 
+    PACKAGE p2 
+  ON 
+    spk2.PACKAGE_REF = p2.PACKAGE_ID
+  INNER JOIN 
+    ATTRIBUTE a2 
+  ON 
+    p2.PACKAGE_ID = a2.PACKAGE_REF
+  WHERE 
+    spk.SESSION_PARTITION_REF = NEW.SESSION_PARTITION_REF
+  AND 
+    spk.PACKAGE_REF = NEW.PACKAGE_REF
+  AND 
+    spk2.ENABLED = true
+  AND 
+    a.CLUSTER_REF = a2.CLUSTER_REF
+  AND 
+    ((a.CODE = a2.CODE AND a.MANUFACTURER_CODE = a2.MANUFACTURER_CODE)
+      OR 
+      (a.CODE = a2.CODE AND a.MANUFACTURER_CODE IS NULL AND a2.MANUFACTURER_CODE IS NULL))
+  AND 
+    p.PACKAGE_ID <> p2.PACKAGE_ID;
+END;
+
+/* Trigger that deals with code conflicts in attributes when a session package is re-enabled */
```

Comment:
How does re-enablement work in for a user? how the user flow will be here in that case?